src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -568,4 +588,54 @@ public EntityCitizen getCitizen()
     {
         return worker;
     }
+
+    private boolean harvestCrop(final BlockPos position)
+    {
+        IBlockState curBlockState = world.getBlockState(position);
+
+        if (!(curBlockState.getBlock() instanceof IGrowable && curBlockState.getBlock() instanceof BlockCrops))
+        {
+            return false;
+        }
+        BlockCrops crops = (BlockCrops) curBlockState.getBlock();
+
+        if (!crops.isMaxAge(curBlockState))
+        {
+            return false;
+        }
+
+        final ItemStack tool = worker.getHeldItemMainhand();
+
+        //calculate fortune enchantment
+        int fortune = Utils.getFortuneOf(tool);
+
+        final List<ItemStack> drops = crops.getDrops(world, position, curBlockState, fortune);
+
+        final Item seedItem = crops.getItemDropped(curBlockState, world.rand, fortune);
+        if (seedItem != null)
+        {
+            for (Iterator<ItemStack> iterator = drops.iterator(); iterator.hasNext();) {
+                final ItemStack drop = iterator.next();
+
+                // Remove a seed, then break.
+                if (! (drop.getItem() == seedItem) || crops instanceof BlockCarrot || crops instanceof BlockPotato) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use the opposite operator ("!=") instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1940)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)